### ISSUE TITLE
ERROR  : Failed to mount squashfs image in (read only): Invalid argument

### ISSUE THREAD

**Issue by JoKalliauer** [apptainer/singularity#5482]
### Version of Singularity:
`2.4.2-dist`

#### Build
[MWE.def](https://github.com/hpcng/singularity/files/5034476/MWE.def.txt)
```sh
BootStrap: docker
From: centos:7

%runscript

    echo "HelloWorld!"
```
`sudo singularity build MWE.img MWE.def `

### run
`singularity run MWE.img`

#### Expected behavior
```
HelloWorld!
```

#### Actual behavior
```sh
ERROR  : Failed to mount squashfs image in (read only): Invalid argument
ABORT  : Retval = 255
```

### What OS/distro are you running

```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Comment by dtrudg** [comment]
2.4.2 is very old and no longer supported. 

The kernel version in Ubuntu 20.04 needs a fix that was introduced in Singularity 3.5.3. Please update to a newer version of Singularity.

**Comment by JoKalliauer** [comment]
@dctrud : Thanks that solved the issue.
